app/assets/javascripts/components/projects/ProjectBoard.js

```diff
@@ -19,89 +17,101 @@ import { removeNotification } from '../../actions/notifications';
 import StorySearch from '../search/StorySearch';
 import SearchResults from './../search/SearchResults';
 import ProjectLoading from './ProjectLoading';
+import SideBar from './SideBar';
+import Columns from '../Columns/Columns';
+import Stories from "../stories/Stories";
+import Sprints from "../stories/Sprints";
+
+export const ProjectBoard = ({
+  fetchProjectBoard,
+  projectId,
+  projectBoard,
+  createStory, 
+  closeHistory,
+  notifications, 
+  removeNotification, 
+  history,
+  chillyBinStories,
+  backlogSprints,
+  doneSprints,
+  fetchPastStories
+}) => {
+  useEffect(() => {
+    fetchProjectBoard(projectId)
+  }, [fetchProjectBoard, projectId]);
 
-export class ProjectBoard extends React.Component {
-  componentWillMount() {
-    this.props.fetchProjectBoard(this.props.projectId);
+  if (!projectBoard.isFetched) {
+    return <ProjectLoading data-id="project-loading" />;
   }
 
-  render() {
-    if (!this.props.projectBoard.isFetched) {
-      return <ProjectLoading data-id="project-loading" />;
+  const columns = [
+    {
+      title: I18n.t("projects.show.chilly_bin"),
+      renderAction: () =>
+        <AddStoryButton
+          onAdd={() => createStory({
+            state: Story.status.UNSCHEDULED
+          })}
+        />,
+      children: <Stories stories={chillyBinStories} />
+    },
+    {
+      title: `${I18n.t("projects.show.backlog")} / ${I18n.t("projects.show.in_progress")}`,
+      renderAction: () =>
+        <AddStoryButton
+          onAdd={() => createStory({
+            state: Story.status.UNSTARTED
+          })}
+        />,
+      children: <Sprints sprints={backlogSprints} />
+    },
+    {
+      title: I18n.t("projects.show.done"),
+      children: <Sprints sprints={doneSprints} fetchStories={fetchPastStories} />
     }
+  ];
 
-    const { 
-      projectId, 
-      createStory, 
-      closeHistory,
-      notifications, 
-      removeNotification, 
-      history,
-      projectBoard
-    } = this.props;
+  return (
+    <div className="ProjectBoard">
+      <StorySearch projectId={projectId} loading={projectBoard.search.loading} />
 
-    return (
-      <div className="ProjectBoard">
-        <StorySearch projectId={projectId} loading={projectBoard.search.loading} />
+      <SideBar />
 
-        <Notifications
-          notifications={notifications}
-          onRemove={removeNotification}
-        />
+      <Notifications
+        notifications={notifications}
+        onRemove={removeNotification}
+      />
 
-        <Column title={I18n.t("projects.show.chilly_bin")}
-          renderAction={() =>
-            <AddStoryButton
-              onAdd={() => createStory({
-                state: Story.status.UNSCHEDULED
-              })}
+      {
+        projectBoard.reverse
+          ? 
+            <Columns
+              data-id="reversed-column"
```

Comment:
Should we avoid duplication here moving the props for this component to the top of the `render` method? Something like this:

```js
const columnProps = {
  'data-id': projectBoard.reverse ? 'reversed-column' : 'normal-column',
  columns; projectBoard.reverse ? columns.reverse() : columns
};

// and then

<Columns {...columnsProps} />
```

Reply:
Nice! Thanks.